signer/src/bitcoin/error.rs

```diff
@@ -0,0 +1,83 @@
+//! Errors for the bitcoin module
+
+/// Errors for Bitcoin Core
+#[derive(Debug, thiserror::Error)]
+pub enum BitcoinError {
```

Comment:
This kind of pattern is a very common pattern and not bad. But I don't think we gain anything from it in this case.

The approach that I've been a proponent of follows what was laid out here https://github.com/stacks-network/sbtc/pull/161#discussion_r1605010772. The TLDR is just make sure to add context to our errors, and keep adding most things to the crate `Error` type so that we can have `anyhow` like simplicity. So far this approach hasn't costs us anything and it makes things really simple.

I'm not sure what this change gains us. So far it seems like we need to add `.into()` in some places where before we didn't before.

Reply:
I'm on the other side of the fence for this one, but I'll revert this for now since it's not necessary for the PR and we can discuss separately

Reply:
Reverted in [af7f73d](https://github.com/stacks-network/sbtc/pull/834/commits/af7f73d952a0be495d26108ccf173230ae6c5399)